Revit_Core_Engine/Query/CurtainWallCurves.cs

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Adapters.Revit;
+using BH.oM.Geometry;
+using BH.oM.Dimensional;
+using BH.Engine.Facade;
+using System;
+using System.Collections.Generic;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static List<ICurve> CurtainWallCurves(this Wall wall)
+        {
+            List<ICurve> result = new List<ICurve>();
+            List<IElement1D> allCrvs = new List<IElement1D>();
+
+            PullGeometryConfig geometryConfig = new PullGeometryConfig();
+            Options geometryOptions = Create.Options(Autodesk.Revit.DB.ViewDetailLevel.Fine, geometryConfig.IncludeNonVisible = true, false);
+            List<Curve> objs = wall.Curves(geometryOptions);
+            foreach (Curve crv in objs)
+            {
+                allCrvs.Add(crv.IFromRevit());
+            }
+            
+            //Get only external curves from all curves
+            foreach (ICurve crv in allCrvs)
+            {
+                List<IElement1D> adjElems = crv.AdjacentElements(allCrvs);
```

Comment:
At the beginning I was surprised why you were using `IElement1D` interface instead of `ICurve` to store `allCrvs`. It all became clear when I looked up the `AdjacentElements` method. There seems to be a design flaw, i.e. the method should not take `List<IElement1D>` as `referenceElements`, it should take `IEnumerable<IElement1D>`. Then you should be able to declare allCrvs as List<ICurve> and implicitly cast it to `IEnumerable<IElement1D>` on method call.

This comment is not related to this PR, rather highlighting an issue with one of the references libraries.

Reply:
Absolutely dead on, this will be captured as a separate BHoM_Engine issue as those methods will need some clean up, good catch.